File: RepoTransBench/target_projects/Java/C#/nzakas_props2js/tests/original/PropertyConverterTest.cs
using System;
using Xunit;
using System.Collections.Generic;

namespace Props2Js.Tests.Original
{
    public class PropertyConverterTest
    {
        [Fact]
        public void TestConvertToJson_withString()
        {
            var p = new Dictionary<string, string> { { "name", "value" } };
            string json = PropertyConverter.ConvertToJson(p);
            Assert.Contains(@"""name"":""value""", json);
        }

        [Fact]
        public void TestConvertToJson_withInt()
        {
            var p = new Dictionary<string, string> { { "intValue", "123" } };
            string json = PropertyConverter.ConvertToJson(p);
            Assert.Contains(@"""intValue"":123", json);
        }

        [Fact]
        public void TestConvertToJson_withFloat()
        {
            var p = new Dictionary<string, string> { { "floatValue", "12.34" } };
            string json = PropertyConverter.ConvertToJson(p);
            Assert.Contains(@"""floatValue"":12.34", json);
        }

        [Fact]
        public void TestConvertToJson_withBooleanTrue()
        {
            var p = new Dictionary<string, string> { { "flag", "true" } };
            string json = PropertyConverter.ConvertToJson(p);
            Assert.Contains(@"""flag"":true", json);
        }

        [Fact]
        public void TestConvertToJson_withBooleanFalse()
        {
            var p = new Dictionary<string, string> { { "flag", "false" } };
            string json = PropertyConverter.ConvertToJson(p);
            Assert.Contains(@"""flag"":false", json);
        }

        [Fact]
        public void TestConvertToJsonP()
        {
            var p = new Dictionary<string, string> { { "a", "1" } };
            string outVal = PropertyConverter.ConvertToJsonP(p, "cb");
            Assert.StartsWith("cb(", outVal);
            Assert.EndsWith(");", outVal.Trim());
        }

        [Fact]
        public void TestConvertToJavaScript_var()
        {
            var p = new Dictionary<string, string> { { "foo", "bar" } };
            string js = PropertyConverter.ConvertToJavaScript(p, "testVar");
            Assert.StartsWith("var testVar=", js);
            Assert.EndsWith(";", js.Trim());
        }

        [Fact]
        public void TestConvertToJavaScript_assignment()
        {
            var p = new Dictionary<string, string> { { "foo", "bar" } };
            string js = PropertyConverter.ConvertToJavaScript(p, "object.property");
            Assert.DoesNotContain("var ", js);
            Assert.StartsWith("object.property=", js);
            Assert.EndsWith(";", js.Trim());
        }

        [Fact]
        public void TestConvertToJson_withMixedTypes()
        {
            var p = new Dictionary<string, string>
            {
                { "string", "text" },
                { "int", "42" },
                { "float", "2.718" },
                { "trueBool", "true" },
                { "falseBool", "false" }
            };
            string json = PropertyConverter.ConvertToJson(p);
            Assert.Contains(@"""string"":""text""", json);
            Assert.Contains(@"""int"":42", json);
            Assert.Contains(@"""float"":2.718", json);
            Assert.Contains(@"""trueBool"":true", json);
            Assert.Contains(@"""falseBool"":false", json);
        }

        [Fact]
        public void TestConvertToJson_handlesEmptyProperties()
        {
            var p = new Dictionary<string, string>();
            string json = PropertyConverter.ConvertToJson(p);
            Assert.Equal("{}", json);
        }
    }
}

File: RepoTransBench/source_projects/Java/nzakas_props2js/tests/net/nczonline/web/props2js/PropertyConverterTest.java
package net.nczonline.web.props2js;

import org.junit.Test;
import static org.junit.Assert.*;
import java.util.Properties;

public class PropertyConverterTest {

    @Test
    public void testConvertToJson_withString() {
        Properties p = new Properties();
        p.setProperty("name", "value");
        String json = PropertyConverter.convertToJson(p);
        assertTrue(json.contains("\"name\":\"value\""));
    }

    @Test
    public void testConvertToJson_withInt() {
        Properties p = new Properties();
        p.setProperty("intValue", "123");
        String json = PropertyConverter.convertToJson(p);
        assertTrue(json.contains("\"intValue\":123"));
    }

    @Test
    public void testConvertToJson_withFloat() {
        Properties p = new Properties();
        p.setProperty("floatValue", "12.34");
        String json = PropertyConverter.convertToJson(p);
        assertTrue(json.contains("\"floatValue\":12.34"));
    }

    @Test
    public void testConvertToJson_withBooleanTrue() {
        Properties p = new Properties();
        p.setProperty("flag", "true");
        String json = PropertyConverter.convertToJson(p);
        assertTrue(json.contains("\"flag\":true"));
    }

    @Test
    public void testConvertToJson_withBooleanFalse() {
        Properties p = new Properties();
        p.setProperty("flag", "false");
        String json = PropertyConverter.convertToJson(p);
        assertTrue(json.contains("\"flag\":false"));
    }

    @Test
    public void testConvertToJsonP() {
        Properties p = new Properties();
        p.setProperty("a", "1");
        String out = PropertyConverter.convertToJsonP(p, "cb");
        assertTrue(out.startsWith("cb("));
        assertTrue(out.endsWith(");"));
    }

    @Test
    public void testConvertToJavaScript_var() {
        Properties p = new Properties();
        p.setProperty("foo", "bar");
        String js = PropertyConverter.convertToJavaScript(p, "testVar");
        assertTrue(js.startsWith("var testVar="));
        assertTrue(js.endsWith(";"));
    }

    @Test
    public void testConvertToJavaScript_assignment() {
        Properties p = new Properties();
        p.setProperty("foo", "bar");
        String js = PropertyConverter.convertToJavaScript(p, "object.property");
        assertFalse(js.startsWith("var "));
        assertTrue(js.startsWith("object.property="));
        assertTrue(js.endsWith(";"));
    }

    @Test
    public void testConvertToJson_withMixedTypes() {
        Properties p = new Properties();
        p.setProperty("string", "text");
        p.setProperty("int", "42");
        p.setProperty("float", "2.718");
        p.setProperty("trueBool", "true");
        p.setProperty("falseBool", "false");
        String json = PropertyConverter.convertToJson(p);
        assertTrue(json.contains("\"string\":\"text\""));
        assertTrue(json.contains("\"int\":42"));
        assertTrue(json.contains("\"float\":2.718"));
        assertTrue(json.contains("\"trueBool\":true"));
        assertTrue(json.contains("\"falseBool\":false"));
    }

    @Test
    public void testConvertToJson_handlesEmptyProperties() {
        Properties p = new Properties();
        String json = PropertyConverter.convertToJson(p);
        assertEquals("{}", json);
    }
}